File: Introductory_Problems/Gray_Code.py
def int_to_binary(num, n) :
    binary = ["0"] * n
    i = 0
    while (num > 0) :
        binary[i] = str(num & 1)
        num >>= 1
        i += 1
        
    return''.join(binary[::-1])


def gray_code(n) :
    for i in range(2**n) :
        print(int_to_binary((i >> 1) ^ i, n))
        
if __name__ == "__main__" :
    n = int(input())
    gray_code(n)

File: Introductory_Problems/Gray_Code.cpp
#include <bits/stdc++.h>
using namespace std;

// convert number to binary 
string intToBin(int num, int n) {
    string bin(n, '0');
    int i = 0;
    while (num > 0) {
        bin[i] = '0' + (num & 1);
        num >>= 1;
        i++;
    }
    reverse(bin.begin(), bin.end());
    return bin;
}

// Formula for graycode num (num ^ num >> 1)
string getGraycode(int num, int n) {
    return intToBin((num)^(num>>1), n);
}

int main() {
    int n;
    cin >> n;
    
    for (int i =0; i < pow(2, n); i++) {
        cout << getGraycode(i, n) << endl;
    }
}